crates/ecs/src/lib.rs

```diff
@@ -789,6 +1077,34 @@ impl<T: Debug + Send + Sync + 'static> ComponentVec for ComponentVecImpl<T> {
     fn remove(&self, index: usize) {
         self.write().expect("Lock is poisoned").swap_remove(index);
     }
+    fn move_element(&self, source_index: usize, target_arch: &Archetype) -> ArchetypeResult<()> {
+        let value = self
+            .write()
+            .expect("Lock is poisoned")
+            .swap_remove(source_index);
+
+        // Only perform move if component type exists in target_arch.
+        if let Some(target_vec) = target_arch
+            .component_typeid_to_component_vec
+            .get(&TypeId::of::<T>())
+        {
+            let mut component_vec = borrow_component_vec_mut::<T>(target_vec.deref())
+                .ok_or(ArchetypeError::CouldNotBorrowComponentVec(TypeId::of::<T>()))?;
+
+            component_vec.push(value);
+        }
+
+        Ok(())
+    }
```

Comment:
If i remember correctly I need the type T to be able to find the correct compvec to move to in the target. Only the comp_vec has access to this generic which is why there was a need for archetype to be sent to the comp_vec once there was a need for specific elements to be moved.

At the moment OK is returned since I wanted a simple way to remove and discard component that had no corresponding type in target archetype. I will try to move this behaviour from comp_vec to make more sense and eliminate instances where Ok is returned while the call was erroneous

Reply:
Now returns error if target_arch does not contain the required component vector